lib/spack/spack/build_systems/python.py

```diff
@@ -447,8 +494,36 @@ def global_options(self, spec, prefix):
         """
         return []
 
+    @property
+    def _build_venv_path(self):
+        """Return the path to the virtual environment used for building when
+        python is external."""
+        return os.path.join(self.spec.package.stage.path, "build_env")
+
+    @property
+    def _build_venv_python(self) -> Executable:
+        """Return the Python executable in the build virtual environment when
+        python is external."""
+        return Executable(os.path.join(self._build_venv_path, "bin", "python"))
```

Comment:
Will it always be `python`? For example, on Ubuntu where only `python3` exists.

Reply:
Since Python 2.7 apparently: https://github.com/python/cpython/commit/7ded1f0f694f0f99252ea19eca18b74ea5e36cb0